cmd/actuary/server/server.go

```diff
@@ -34,68 +37,93 @@ var (
+			ctx := context.Background()
+			cli, err := client.NewEnvClient()
+			if err != nil {
+				log.Fatalf("Could not create new client: %s", err)
+			}
+			nodeList, err := cli.NodeList(ctx, types.NodeListOptions{})
+			if err != nil {
+				log.Fatalf("Could not get list of nodes: %s", err)
+			}
+
+			// Send official list of nodes from docker client to browser
+			mux.HandleFunc("/getNodeList", func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "text/html")
+				w.WriteHeader(http.StatusOK)
+				var b bytes.Buffer
+				for _, node := range nodeList {
+					b.Write([]byte(node.ID + " "))
+				}
+				b.WriteTo(w)
+			})
+
+			// Where nodes send DATA from check.go, where javascript requests receives specific node DATA
+			mux.HandleFunc("/results", func(w http.ResponseWriter, r *http.Request) {
 				if r.Method == "POST" {
 					output, err := ioutil.ReadAll(r.Body)
 					if err != nil {
 						log.Fatalf("Error reading: %s", err)
 					}
-					// Get nodeID, results passed in from check on a particular node
 					var req check.Request
 					err = json.Unmarshal(output, &req)
 					if err != nil {
 						log.Fatalf("Error unmarshalling id: %s", err)
 					}
 					reqList = append(reqList, req)
 					nodeID := string(req.NodeID)
-					log.Printf("NODE ID %v", nodeID)
 					results := req.Results
 					report.Mu.Lock()
 					report.Outputs[nodeID] = results
 					report.Mu.Unlock()
+				} else if r.Method == "GET" {
+					nodeID := r.URL.Query().Get("nodeID")
+					if nodeID != "" {
+						w.Header().Set("Content-Type", "application/json")
+						w.WriteHeader(http.StatusOK)
+						report.Mu.Lock()
+						w.Write(report.Outputs[nodeID])
+						report.Mu.Unlock()
+					} else {
+						log.Fatalf("Node ID not entered")
+					}
 				}
 			})
 
+			// Path to return css/js/html
 			mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-				index := strings.Split(r.URL.Path, "/")
-				if index[len(index)-1] != "" {
-					w.Header().Set("Content-Type", "application/json") //change to writeHeader!
-					report.Mu.Lock()
-					w.Write(report.Outputs[index[len(index)-1]])
-					report.Mu.Unlock()
-				} else {
-					log.Fatalf("Node ID not entered")
+				currentDir, err := os.Getwd()
+				if err != nil {
+					log.Fatalf("Error getting current directory: %s", err)
 				}
-			})
-
-			mux.HandleFunc("/results/", func(w http.ResponseWriter, r *http.Request) {
-				path := htmlPath
-				handler := http.StripPrefix("/results/", http.FileServer(http.Dir(path)))
+				path := filepath.Join(currentDir, htmlPath)
+				handler := http.FileServer(http.Dir(path))
 				handler.ServeHTTP(w, r)
 			})
 
-			mux.HandleFunc("/getNodes", func(w http.ResponseWriter, r *http.Request) {
+			// Determine whether or not a specified node has been processed -- ie if its results are ready to be displayed
+			mux.HandleFunc("/checkNode", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why do we need a `/checkNode`? Maybe we can add a parameter to the `/results` GET that is a filter (filter down to certain node ID), if efficiency is what you're concerned about?